ulib/experimental/Steel.SelEffect.fsti

```diff
@@ -271,6 +289,15 @@ val change_slprop_rel (p q:vprop)
       rel (sel_of p m) (sel_of q m))
   ) : SteelSel unit p (fun _ -> q) (fun _ -> True) (fun h0 _ h1 -> rel (h0 p) (h1 q))
 
+val change_slprop_rel_with_cond (p q:vprop)
```

Comment:
I'm not sure you need this. Calling change_slprop_rel with a lemma containing more preconditions is supposed to push these preconditions (corresponding to your cond argument here) into the context.

Reply:
Am I really allowed to call `change_slprop_rel` with a lemma with more preconditions than just `interp p m`? That lemma is supposed to hold for any such m, but in my case I can rewrite only under `cond`.

For instance, for `vdep`, I can do the vprop rewrite only if the "payload" vprop corresponds to the case indexed by the "key". There I couldn't use `change_slprop_rel`, which rewrites unconditionally.

Reply:
Yes, see for instance https://github.com/FStarLang/FStar/blob/steel/examples/steel/Selectors.Tree.Core.fst#L134.
The extra condition is_null_t is not included in the signature of change_slprop_rel; it is added to the context, and proven thanks to the precondition of the function.

Reply:
This example involves a precondition that does not depend on the memory state. In my case, I mean preconditions that _do_ depend on the memory state --- namely the return value of a selector in the current memory state --- thus are not captured by the lemma. I claim that `change_slprop_rel` cannot handle those.